src/fetchers/tarball-fetcher.js

```diff
@@ -189,7 +189,12 @@ export default class TarballFetcher extends BaseFetcher {
       return resolve({
         hash: this.hash || hexDigest,
       });
+    };
+
+    extractorStream.pipe(untarStream).on('finish', () => {
+      setTimeout(end, 1000);
```

Comment:
This is the delay I was mentionning. Removing it seems to cause issues with:

```
D:\yarn\bin\yarn.cmd jest -t "it should install the dependencies of any dependency fetched from the filesystem"
```

It seems like the `package.json` for this package isn't flushed until after the `finish` event is emitted, which causes issues when we try to read the dependencies later on. Yarn fallback to "no dependency", which makes the test crash because it test that the dependency can be loaded.